.ci/Jenkinsfile

```diff
@@ -258,3 +299,8 @@ pipeline {
     }
   }
 }
+
```

Comment:
```suggestion
def ubuntuPod(){
 return { yamlFile 'resources/k8s/UbuntuPod.yml' }
}
```

Reply:
this implemented as step (e.g. ubuntuPod()) or substep(e.g. pods.ubuntu()) will make sure easy to control all the k8s agents

Reply:
a Map can also make the thing (pods.get('ubuntu'))

pods.groovy
```
def get(name){
  return pods[name]
}
```

Reply:
I rather prefer to use the standard approach to define the k8s agent, let me explain the reason:

The k8s definition will be stored in each github project, therefore the shared library step will be a quite opinionated approach, while I don't see the benefits of using it, but adding certain indirections to finally call a relative path to a pods definition stored in the project. 

```
agent { kubernetes { yamlFile 'resources/k8s/UbuntuPod.yml' } }
```

vs

```
agent { kubernetes ubuntuPod() }
```

even the first one is more verbose, it reflects what's the agent and where the definition is stored, while the later is somehow more descriptive but it hides where the pod definition is stored.